app/src/main/java/com/sun/ev_dictionary/ui/splash/SplashViewModel.kt

```diff
@@ -0,0 +1,67 @@
+package com.sun.ev_dictionary.ui.splash
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.sun.ev_dictionary.base.BaseViewModel
+import com.sun.ev_dictionary.data.source.repository.EnglishWordsRepository
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+
+class SplashViewModel(
+    private val englishWordsRepository: EnglishWordsRepository
+) : BaseViewModel() {
+
+    private val compositeDisposable = CompositeDisposable()
+    private var _countProgress = 0
+    private val _numberOfWordsInserted = MutableLiveData<Int>()
+    val numberOfWordsInserted: LiveData<Int>
+        get() = _numberOfWordsInserted
+    private val _numberOfWords = MutableLiveData<Int>()
+    val numberOfWords: LiveData<Int>
+        get() = _numberOfWords
+    private val _isWordInsertionFinished = MutableLiveData<Boolean>()
+    val isWordsInsertionFinished: LiveData<Boolean>
+        get() = _isWordInsertionFinished
+
+    init {
+        _isWordInsertionFinished.value = false
+        _numberOfWordsInserted.value = 0
+    }
+
+    override fun onStart() {
+        getData()
+        getNumberOfWords()
+    }
+
+    override fun onStop() {
+        compositeDisposable.clear()
+    }
+
+    private fun getData() {
+        compositeDisposable.add(
+            englishWordsRepository.getWordsFromTextFile()
+                .doOnNext { progress -> increaseInsertedWords(progress) }
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({ onInsertFinish() },
+                    { throwable -> throwable.localizedMessage })
+        )
+    }
+
+    private fun getNumberOfWords() {
+        _numberOfWords.value = englishWordsRepository.getNumberOfWords()
```

Comment:
các hoạt động uơng tác với db em đã tạo trên luồng mới chưa nhỉ

Reply:
cái này là trong lúc đọc file e đã get rồi ạ. Ở luồng của fromCallable ấy ạ
Ở khi e chỉ gán lại giá trị khi mà nó được phá ra thôi ạ.
// ở trong LocalDataSource e làm như dưới ạ:
   override fun getNumberOfWords(): Int = _numberOfWords

    override fun getWordsFromTextFile(): Observable<Int> {
        return Observable.fromCallable {
            getLines(bufferedReader)
        }
            .flatMapIterable { lines -> lines }
            .map { line -> convertLineToEnglishWord(line) }
            .map { englishWord -> insertWordsToDatabase(englishWord) }
    }

    private fun getLines(bufferedReader: BufferedReader): List<String> {
        bufferedReader.useLines { lines ->
            return lines.toList().also { _numberOfWords = it.size }
        }
    }